### PR TITLE
DEV-5358 | Better management of subscription statuses for stripe transactions import

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

Adds optionality to the stripe transactions import command such that:

- Can exclude one-times
- Can exclude recurring
- Can specify subscription status

Additionally, we now default to retrieving all subscription statuses when retrieving subscriptions (formerly, we went with Stripe defaults, which exclude canceled, and that led to missing contributions).

In a follow up ITS ticket, we'll run the command focusing on canceled recurring subscriptions. 

#### Why are we doing this? How does it help us?

Get missing contributions in revengine

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

yes/existing ones updated.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

- [DEV-5358](https://news-revenue-hub.atlassian.net/browse/DEV-5358)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

[ITS-2624](https://news-revenue-hub.atlassian.net/browse/ITS-2624)

[DEV-5358]: https://news-revenue-hub.atlassian.net/browse/DEV-5358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by njharman — APPROVED** [review]

File: apps/contributions/management/commands/import_stripe_transactions_data.py [hunk 1]
```diff
@@ -59,6 +59,15 @@ def add_arguments(self, parser: CommandParser) -> None:
         )
         parser.add_argument("--suppress-stripe-info-logs", action="store_true", default=False)
         parser.add_argument("--sentry-profiler", action="store_true", default=False)
+        parser.add_argument("--exclude-one-times", action="store_true", default=False)
+        parser.add_argument("--exclude-recurring", action="store_true", default=False)
+        # see https://docs.stripe.com/api/subscriptions/list#list_subscriptions-status for available values.
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply apps/contributions/management/commands/import_stripe_transactions_data.py:71]: 7359998

File: apps/contributions/management/commands/import_stripe_transactions_data.py [hunk 2]
```diff
@@ -59,6 +59,15 @@ def add_arguments(self, parser: CommandParser) -> None:
...
+        # Note that "uncanceled" is not a valid value, but we use it to indicate that no value should be sent to
+        # Stripe when retrieving subscriptions, which results in default behavior of all being returned that are not
+        # canceled.
+        parser.add_argument(
+            "--subscription-status", type=str, default="all", choices=["all", "ended", "canceled", "uncanceled"]
+        )  # need to limit to acceptable values
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply apps/contributions/management/commands/import_stripe_transactions_data.py]: 7359998

File: apps/contributions/stripe_import.py [hunk 3]
```diff
@@ -278,6 +278,13 @@ class StripeTransactionsImporter:
     to_date: datetime.datetime = None
     retrieve_payment_method: bool = False
     sentry_profiler: bool = False
+    include_one_time_contributions: bool = True
+    include_recurring_contributions: bool = True
+    # # see https://docs.stripe.com/api/subscriptions/list#list_subscriptions-status for available values.
```
[the remark this answers is not in the document]
Comment (benjaminEwhite) [reply apps/contributions/stripe_import.py]: 7359998

**Review by x110dc — APPROVED** [review]
lgtm

**Pull request merged**